Mage.Server/src/main/java/mage/server/util/ThreadExecutor.java

```diff
@@ -1,42 +1,39 @@
-
 package mage.server.util;
 
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.ScheduledExecutorService;
-import java.util.concurrent.ThreadFactory;
-import java.util.concurrent.ThreadPoolExecutor;
-import java.util.concurrent.TimeUnit;
+import mage.server.managers.IConfigSettings;
+import mage.server.managers.IThreadExecutor;
+
+import java.util.concurrent.*;
 
 /**
- *
  * @author BetaSteward_at_googlemail.com
  */
-public enum ThreadExecutor {
-instance;
-    private static final ExecutorService callExecutor = Executors.newCachedThreadPool();
-    private static final ExecutorService userExecutor = Executors.newCachedThreadPool();
-    private static final ExecutorService gameExecutor = Executors.newFixedThreadPool(ConfigSettings.instance.getMaxGameThreads());
-    private static final ScheduledExecutorService timeoutExecutor = Executors.newScheduledThreadPool(4);
-    private static final ScheduledExecutorService timeoutIdleExecutor = Executors.newScheduledThreadPool(4);
+public class ThreadExecutor implements IThreadExecutor {
+    private final ExecutorService callExecutor;
+    private final ExecutorService gameExecutor;
+    private final ScheduledExecutorService timeoutExecutor;
+    private final ScheduledExecutorService timeoutIdleExecutor;
```

Comment:
Initialisation is delayed at instance construction to allow to use the injected configuration.